src/impl_constructors.rs

```diff
@@ -252,3 +258,28 @@ impl<S, A, D> ArrayBase<S, D>
     }
 
 }
+
+use super::NdFloat;
+
+impl<S, A, D> ArrayBase<S, D>
+    where S: DataOwned<Elem=A>,
+          A: NdFloat+AsPrim,
+          D: Dimension,
+{
+    pub fn randn_rng<R: Rng>(dim: D, rng: &mut R) -> ArrayBase<Vec<A>, D>
+    {
+        let size = dim.size_checked().expect("Shape too large: overflow in size");
+        let normal = Normal::new(1.0, 0.0);
+        let iter = (0..size).map(|_| normal.ind_sample(rng).as_());
+        let arr = OwnedArray::from_iter(iter);
+        arr.into_shape(dim).unwrap()
+    }
+
+    pub fn randn(dim: D) -> ArrayBase<Vec<A>, D>
```

Comment:
Because this might be generally useful: Instead of `ArrayBase<Vec<A>, D>` this should just be `OwnedArray<A, D>`.

But the impl block is written for all `ArrayBase` with owned storage, and the type parameter `S` is unused. So I think the intention is to write `ArrayBase<S, D>` as the return type here instead. This makes it work for both OwnedArray and RcArray.